viewing_party/main.py

```diff
@@ -0,0 +1,247 @@
+    outputs: a list of movie dictionaries unique to that user (a suggested watchlist to give to their friends) 
+    """
+
+    # initialize unique titles list for movie dictionaries 
+    users_unique_movies = []
+    friends_movies = []
+
+    # should return an empty list if user's list is empty?
+    if not user_data["watched"]:
+        return []
+
+    # add all friends watched movies to a new list
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_movies.append(movie)
+
+    # for each movie in user's watched list
+    for movie in user_data["watched"]:
+        # check to see if the "title" value matches any in each friend's watched list, if not:
+        if movie not in friends_movies:
+            # add the DICTIONARY for the user's movie to unique titles list
+            users_unique_movies.append(movie)
+
+    # return a list of the movies the user has watched that their friends haven't seen yet
+    return users_unique_movies
+
+def get_friends_unique_watched(user_data):
+
+    friends_unique_movies = []
+    users_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in users_movies and movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+            
+    return friends_unique_movies
+
+#4============================================================================
+
+def get_available_recs(user_data):
+    """
+    inputs: the user_data dictionary
+    outputs: a list of movie dictionaries that the user hasn't seen yet, is in their host network, and that at least one friend has seen; if no subscription data is gievn, it will just give movies the user hasn't seen and at least one friend has seen
+    """
+
+    recommended_movs = []
+    users_movies = []
+    
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+        
+    subscriptions_key = "subscriptions"
+
+    # for friend in user_data["friends"]:
+    #     for movie in friend["watched"]:
+    #         if movie not in recommended_movs:
+    #             if subscriptions_key not in user_data:
+    #                 recommended_movs.append(movie)
+    #             elif movie["host"] in user_data["subscriptions"]:
+    #                 recommended_movs.append(movie)    
+    # return recommended_movs
+
+    for friend in user_data["friends"]:
+        if not friend["watched"]:
+            return []
+        else:
+            for movie in friend["watched"]:
+                if movie not in recommended_movs:
+                    if subscriptions_key not in user_data:
+                        recommended_movs.append(movie)
+                    elif movie["host"] in user_data["subscriptions"]:
+                        recommended_movs.append(movie)    
```

Comment:
The nesting here is a little confusing. Consider using a compound conditional. I don't think this conditional test is needed per the requirements `subscriptions_key not in user_data:`: 
```suggestion
                if movie not in recommended_movs and movie["host"] in user_data["subscriptions"]:
                        recommended_movs.append(movie)
```